### PR TITLE
sunshine: 0.21.0 -> 0.22.0

### LINKED ISSUE #293917: Update request: sunshine 0.21.0 → 0.22.0
- Package name: sunshine
- Latest released version: 0.22.0

- Current version on the unstable channel: 0.21.0
- Current version on the stable/release channel: 0.21.0

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@devusb 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by devusb**
## Description of changes

https://github.com/LizardByte/Sunshine/compare/v0.21.0...v0.22.0
- add patches to remove `npm install` and revert upstream commit that tries to copy `udev` and `systemd` files into their store paths
- add scope of https://github.com/NixOS/nixpkgs/pull/235655/

Closes #293917

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by anthr76 — APPROVED** [review]
This lines up with the overlay I've been using.

**Comment by peperunas** [comment]
I also tested the PR and it works perfectly 👍

File: pkgs/servers/sunshine/default.nix [hunk 1]
```diff
@@ -39,29 +39,38 @@
...
+    sha256 = "sha256-O9U4zP1o6yWtzk+2DW7ueimvsTuajLY8IETlvCT4jTE=";
     fetchSubmodules = true;
   };
 
-  # fetch node_modules needed for webui
+  patches = [
+    # remove npm install as it needs internet access -- handled separately below
+    ./dont-build-webui.patch
+    # revert https://github.com/LizardByte/Sunshine/pull/2046 - let cmake install handle udev and systemd files
+    ./systemd-udev-path.patch
```
Comment (pbsds) [inline pkgs/servers/sunshine/default.nix:66]: Can this be a `fetchpatch` with `revert = true`?

Comment (devusb) [reply pkgs/servers/sunshine/default.nix]: @pbsds absolutely! I had completely missed that this was even a thing -- I have needed it so many times. updated. thanks!

**Comment by pbsds** [comment]
```shell
[nix-shell:/dev/shm/nixpkgs-review/pr-293938]$ cat results/sunshine/share/applications/sunshine.desktop | grep ^Exec|  head -n1
Exec=/usr/bin/env systemctl start --u sunshine
```

bump LGTM, but could you patch the desktop entry to not assume FHS? Assume `systemctl` is available in `PATH`, because a nix-store path will cause other problems

**Comment by devusb** [comment]
> ```shell
> [nix-shell:/dev/shm/nixpkgs-review/pr-293938]$ cat results/sunshine/share/applications/sunshine.desktop | grep ^Exec|  head -n1
> Exec=/usr/bin/env systemctl start --u sunshine
> ```
> 
> bump LGTM, but could you patch the desktop entry to not assume FHS? Assume `systemctl` is available in `PATH`, because a nix-store path will cause other problems

Oof, I didn't even notice that -- looks like it slipped in here: https://github.com/LizardByte/Sunshine/pull/1768
I just substituted to switch the desktop entry back to running without `systemd` like it did before -- separately working on a module [here](https://github.com/NixOS/nixpkgs/pull/294641) to actually run it as a daemon.

**Pull request merged**